tests/Io/FileSystemTest.php

```diff
@@ -42,6 +42,132 @@ public function tearDown()
         $this->exec('rm -rf ' . $this->tmp);
     }
 
+    /**
+     * @expectedException        Brick\Std\Io\IoException
+     * @expectedExceptionMessage Error getting real path of invalid_path, check that the path exists
+     */
+    public function testGetRealPathWithInvalidPath()
+    {
+        FileSystem::getRealPath('invalid_path');
+    }
+
+    public function testGetRealPath()
+    {
+        $filePath = __DIR__ . '/../../composer.json';
+        $expectedRealPath = realpath($filePath);
+
+        $this->assertSame($expectedRealPath, FileSystem::getRealPath($filePath));
+    }
+
+    public function testWriteWithAppendFlag()
+    {
+        FileSystem::write('./temp_file', 'data1' . PHP_EOL);
+        FileSystem::write('./temp_file', 'data2', true);
+
+        $this->assertSame('data1' . PHP_EOL . 'data2', FileSystem::read('./temp_file'));
+    }
+
+    public function testWriteWithLockFlag()
+    {
+        $this->assertSame(5, FileSystem::write('./temp_lock_file', '12345', false, true));
+    }
+
+    public function testReadWithMaxLength()
+    {
+        FileSystem::write('./temp_lock_file', 'data');
+
+        $this->assertSame('dat', FileSystem::read('./temp_lock_file', 0, 3));
+    }
+
+    /**
+     * @expectedException        Brick\Std\Io\IoException
+     * @expectedExceptionMessage Error copying ./temp_lock_file to /temp_lock_file
+     */
+    public function testCopyShouldThrowIOException()
+    {
+        FileSystem::write('./temp_lock_file', 'data');        
+        FileSystem::copy('./temp_lock_file', '/temp_lock_file');
```

Comment:
You should not expect that the test will not be run as root to fail.
Instead, you can write to 'non-existent-dir/temp_lock_file'!